app/bundles/EmailBundle/Tests/Controller/Api/EmailApiControllerFunctionalTest.php

```diff
@@ -203,4 +219,232 @@ public function testReplyAction(): void
         $this->assertSame('1', $fetchedStatData['stats'][0]['is_read']);
         $this->assertMatchesRegularExpression('/\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}/', $fetchedStatData['stats'][0]['date_read']);
     }
+
+    public function testSendAction(): void
+    {
+        // Create a user (to test use onwer as mailer):
+        $payload = [
+            'username'      => 'apitest',
+            'firstName'     => 'John',
+            'lastName'      => 'Doe',
+            'email'         => 'john@api.test',
+            'plainPassword' => [
+                'password'  => 'password',
+                'confirm'   => 'password',
+            ],
+            'role'          => 1,
+        ];
+        $this->client->request('POST', '/api/users/new', $payload);
+        $clientResponse = $this->client->getResponse();
+
+        $this->assertSame(201, $clientResponse->getStatusCode(), $clientResponse->getContent());
+
+        $userResponse   = json_decode($clientResponse->getContent(), true);
+        $userId         = $userResponse['user']['id'];
+
+        $this->assertGreaterThan(0, $userId, $clientResponse->getContent());
+
+        // Create a contact:
+        $payload = [
+            'firstname'          => 'Jane',
+            'lastname'           => 'Doe',
+            'email'              => 'jane@api.test',
+            'overwriteWithBlank' => true,
+            'owner'              => $userId,
+        ];
+
+        $this->client->request('POST', '/api/contacts/new', $payload);
```

Comment:
Creating entities needed for the test itself via EntityManager is around 10 times faster than via API. Do you think this and other entities created like this could be rewritten? We've done several rounds of speeding up tests already and I don't want to merge something we'll have to refactor in the future because the test run time will be unbearable.

Reply:
Sure. I noticed the use of the EntityManager when exploring the test facility. I will post a commit shortly.

Reply:
I have changed the test cases to use the entity manager.